### PR TITLE
Added support for Istio CNI

### LINKED ISSUE #2851: istio mTLS causes issue
Mirantis Kubernetes Engine k8s 1.24
istio mesh with istio-cni

> Health checks are received well and plain text by mirrord, but it also receives some TLS traffic that looks destined to the mTLS/istio (i.e before TLS termination)

### PR DESCRIPTION
**Pull request by aviramha**
Closes #2851
Scenarios tested:

1. exec into pod and curl localhost -> stolen
2. via istiogateway -> stolen
3. port forward to pod -> stolen

### CODE REVIEW

File: mirrord/kube/src/api/container.rs [hunk 1]
```diff
@@ -125,18 +126,33 @@ pub fn check_mesh_vendor(pod: &Pod) -> Option<MeshVendor> {
...
+                .iter(),
+        )
+        .map(|status| status.name.as_str())
+        .collect::<Vec<&str>>();
+
+    // check that all the containers are present
+    // we had a case where istio cni was detected as istio while
+    // the init was only present.
+    // leave old logic for detection to not break existing setups
+    if ISTIO_CNI.iter().all(|name| container_names.contains(name)) {
```
Comment (DmitryDodzin) [inline mirrord/kube/src/api/container.rs:145]: Is there a reason for contains? like are the container names different from the lookup? (container named `my-custom-istio-validation` will pass this filter) I'd maybe do a straight up compare

**Review by DmitryDodzin — APPROVED** [review]
LGTM, have a nitpick on container name specificity

File: mirrord/kube/src/api/container.rs (hunk 1 above)
Comment (aviramha) [reply mirrord/kube/src/api/container.rs:145]: Nice catch!

Comment (aviramha) [reply mirrord/kube/src/api/container.rs:145]: Ah sorry, `container_names` is a Vec of names

Comment (aviramha) [reply mirrord/kube/src/api/container.rs:145]: meaning, it does exact match

**Pull request merged**